instance/models/openedx_appserver.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app models - Open EdX AppServer models
+"""
+from django.conf import settings
+from django.db import models
+from django.template import loader
+from django.utils.text import slugify
+
+from instance import ansible
+from instance.logging import log_exception
+from pr_watch.github import get_username_list_from_team
+from .appserver import AppServer
+from .mixins.ansible import AnsibleAppServerMixin, Playbook
+from .mixins.utilities import EmailMixin
+from .utils import SteadyStateException
+
+# Constants ###################################################################
+
+PROTOCOL_CHOICES = (
+    ('http', 'HTTP - Unencrypted clear text'),
+    ('https', 'HTTPS - Encrypted'),
+)
+
+# Models ######################################################################
+
+
+class OpenEdXAppConfiguration(models.Model):
+    """
+    Configuration fields used by OpenEdX Instance and AppServer.
+
+    Mutable on the instance but immutable on the AppServer.
+    """
+    class Meta:
+        abstract = True
+
+    email = models.EmailField(default='contact@example.com', help_text=(
+        'The default contact email for this instance; also used as the from address for emails '
+        'sent by the server.'
+    ))
+    protocol = models.CharField(max_length=5, default='http', choices=PROTOCOL_CHOICES)
+
+    # Ansible-specific settings:
+    configuration_source_repo_url = models.URLField(max_length=256, blank=False)
+    configuration_version = models.CharField(max_length=50, blank=False)
+    configuration_extra_settings = models.TextField(blank=True, help_text="YAML config vars that override all others")
+
+    edx_platform_repository_url = models.CharField(max_length=256, blank=False, help_text=(
+        'URL to the edx-platform repository to use. Leave blank for default.'
```

Comment:
@itsjeyd Turns out I can't mention the default value in `help_text` because the help text becomes part of the migration, so changing the setting would trigger the creation of new migrations :(